cmd/internal/cli/pull.go

```diff
@@ -223,160 +206,46 @@ func pullRun(cmd *cobra.Command, args []string) {
-			imageSigned, err := signing.IsSigned(name, KeyServerURL, 0, false, authToken, true)
-			if err != nil {
-				// err will be: "unable to verify container: %v", err
-				sylog.Warningf("%v", err)
-				// if theres a warning, exit 1
-				exitStat = 1
-			}
-			// if container is not signed, print a warning
-			if !imageSigned {
-				fmt.Fprintf(os.Stderr, "This image is not signed, and thus its contents cannot be verified.\n")
-				resp, err := sypgp.AskQuestion("Do you with to proceed? [N/y] ")
-				if err != nil {
-					sylog.Fatalf("unable to parse input: %v", err)
-				}
-				if resp == "" || resp != "y" && resp != "Y" {
-					fmt.Fprintf(os.Stderr, "Aborting.\n")
-					err := os.Remove(name)
-					if err != nil {
-						sylog.Fatalf("Unabel to delete the container: %v", err)
-					}
-					// exit status 10 after replying no
-					exitStat = 10
-				}
-			}
-		} else {
-			sylog.Warningf("Skipping container verification")
+		err := singularity.LibraryPull(name, ref, transport, args[i], PullLibraryURI, KeyServerURL, authToken, force, unauthenticatedPull)
+		if err == singularity.ErrLibraryUnsigned {
+			exitStat = 1
+		} else if err == singularity.ErrLibraryPullAbort {
+			exitStat = 10
+		} else if err != nil {
+			sylog.Fatalf("%v", err)
 		}
-		fmt.Printf("Download complete: %s\n", name)
-
 	case ShubProtocol:
 		err := shub.DownloadImage(name, args[i], force, noHTTPS)
 		if err != nil {
 			sylog.Fatalf("%v\n", err)
 		}
 	case OrasProtocol:
-		ref = strings.TrimPrefix(ref, "//")
-
-		spec, err := reference.Parse(ref)
-		if err != nil {
-			sylog.Fatalf("Unable to parse oci reference: %s", err)
-		}
-
-		// append default tag if no object exists
-		if spec.Object == "" {
-			spec.Object = SifDefaultTag
-			sylog.Infof("No tag or digest found, using default: %s", SifDefaultTag)
-		}
-
 		ociAuth, err := makeDockerCredentials(cmd)
 		if err != nil {
 			sylog.Fatalf("Unable to make docker oci credentials: %s", err)
 		}
 
-		credFn := func(_ string) (string, string, error) {
-			return ociAuth.Username, ociAuth.Password, nil
-		}
-		resolver := docker.NewResolver(docker.ResolverOptions{Credentials: credFn})
-
-		wd, err := os.Getwd()
-		if err != nil {
-			sylog.Fatalf("Failed to get working directory: %s", err)
-		}
-
-		store := content.NewFileStore(wd)
-		defer store.Close()
-
-		store.AllowPathTraversalOnWrite = true
-		store.DisableOverwrite = !force
-
-		allowedMediaTypes := oras.WithAllowedMediaTypes([]string{SifLayerMediaType})
-		handlerFunc := func(ctx context.Context, desc ocispec.Descriptor) ([]ocispec.Descriptor, error) {
-			if desc.MediaType == SifLayerMediaType {
-				nameOld, _ := content.ResolveName(desc)
-				_ = store.MapPath(nameOld, name)
-			}
-			return nil, nil
-		}
-		pullHandler := oras.WithPullBaseHandler(images.HandlerFunc(handlerFunc))
-
-		_, _, err = oras.Pull(orasctx.Background(), resolver, spec.String(), store, allowedMediaTypes, pullHandler)
+		err = singularity.OrasPull(name, ref, force, ociAuth)
```

Comment:
If go will let you...

```suggestion
		err := singularity.OrasPull(name, ref, force, ociAuth)
```

Reply:
`err` already exists in this scope, I could embed these calls in an `if` statement to limit the scope, but that would mean some pretty long lines